Unit Tests Fail With uv Package Manager

The unit tests are failing in [GitHub actions](https://github.com/ssl-hep/ServiceX_frontend/actions/runs/12729258014/job/35480829476#step:6:210)

I couldn't reproduce on my laptop using pip install, but when I switched to uv, they started showing up.

```
pyenv shell 3.11.14
python -m venv .venv3.11
source .venv3.11/bin/activate
pip install uv
pytest
```

results in 

```
======================================================================= short test summary info =======================================================================
FAILED tests/test_databinder.py::test_yaml_include - ruamel.yaml.composer.ComposerError: found undefined alias 'DEF_query'
======================================================= 1 failed, 163 passed, 279 warnings in 66.74s (0:01:06) ========================================================
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -49,7 +49,7 @@ dependencies = [
     "rich>=13.0.0",  # databinder
     "aiofile",  # compatible versions controlled through miniopy-async
     "make-it-sync",  # compatible versions controlled through func_adl
-    "ruamel.yaml>=0.18",
+    "ruamel.yaml>=0.18,<0.18.7",  # FIXME: ruamel-yaml v0.18.7 has a breaking API change relative to v0.18.6
     "ccorp-yaml-include-relative-path>=0.0.4",
     "filelock>=3.12.0",
     "tenacity >= 9.0.0"

